File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> toAddIndex;
        vector<int> toReturn(2);
        for (int i = 0; i < nums.size(); i++) {
            if (toAddIndex.count(target - nums[i])) {
                toReturn[0] = toAddIndex[target - nums[i]];
                toReturn[1] = i;
                return toReturn;
            }
            toAddIndex[nums[i]] = i;
        }

        return toReturn;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans = [1, len(numbers)]
        currSum = numbers[ans[0] - 1] + numbers[ans[1] - 1]
        while currSum != target:
            if currSum < target:
                ans[0] += 1
            else:
                ans[1] -= 1
            currSum = numbers[ans[0] - 1] + numbers[ans[1] - 1]

        return ans